.github/workflows/test.yml

```diff
@@ -12,64 +13,76 @@ on:
       - 'README.md'
       - 'CHANGELOG.md'
       - 'website/*'
+
 jobs:
+
   build:
     name: Build
     runs-on: ubuntu-latest
     timeout-minutes: 5
+
+    strategy:
+      matrix:
+        go-version: [ '1.18' ]
+
     steps:
-    - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
-    - name: Set up Go
-      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
-      with:
-        go-version-file: 'go.mod'
-      id: go
 
-    - name: Go fmt
-      run: |
-        make fmt
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Run linters
+        uses: golangci/golangci-lint-action@08e2f20817b15149a52b5b3ebe7de50aff2ba8c5 # v3.4.0
+        with:
+          version: latest
 
-    - name: Go vet
-      run: |
-        make vet
+      - name: Generate
+        run: make generate
 
-    - name: Build
-      run: |
-        go build -v .
+      - name: Confirm no diff
+        run: |
+          git diff --compact-summary --exit-code || \
+            (echo "*** Unexpected differences after code generation. Run 'make generate' and commit."; exit 1)
 
+      - name: Build
+        run: make build
 
-# run acceptance tests in a matrix with Terraform core versions
   test:
-    name: Matrix Test
+    name: 'Acc. Tests (OS: ${{ matrix.os }} / TF: ${{ matrix.terraform }})'
     needs: build
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 15
+
     strategy:
       fail-fast: false
       matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
         terraform:
           - '0.12.*'
           - '0.13.*'
           - '0.14.*'
           - '0.15.*'
           - '1.0.*'
           - '1.1.*'
+          - '1.2.*'
+          - '1.3.*'
+
     steps:
-    - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
-    - name: Set up Go
-      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
-      with:
-        go-version-file: 'go.mod'
-      id: go
-    - uses: hashicorp/setup-terraform@633666f66e0061ca3b725c73b2ec20cd13a8fdd1 # v2.0.3
-      with:
-        terraform_version: ${{ matrix.terraform }}
-        terraform_wrapper: false
-    # See also: https://github.com/hashicorp/setup-terraform/issues/143
-    - run: terraform version
-    - run: go mod download
-    - name: TF acceptance tests
-      timeout-minutes: 10
-      env:
-        TF_ACC: "1"
-      run: go test -v -cover ./internal/provider/
+
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version-file: 'go.mod'
```

Comment:
Looks like the tests are running with Go version as specified in `go.mod` whereas `build` phase is using Go version specified in `strategy: matrix`. Just wanted to check that this is intentional?

Reply:
Good callout! I don't think we need the matrix anymore since we are only testing against `1.18` anyways.